File: BACK/nest-app/src/post/post.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { PostService } from './post.service';

@Controller('posts')
export class PostController {
  constructor(private readonly postService: PostService) {}

  @Get()
  async getPostsByCategory(
    @Query('category') categoryId: string,
    @Query('tag') hashTagId: string,
    @Query('page') page: string = '1',
    @Query('pageSize') pageSize: string = '10',
  ) {
    return this.postService.getPosts(
      categoryId ? BigInt(categoryId) : null,
      hashTagId ? BigInt(hashTagId) : null,
      parseInt(page, 10),
      parseInt(pageSize, 10),
    );
  }
}

File: BACK/spring-app/src/main/java/com/starchive/springapp/post/controller/PostController.java
package com.starchive.springapp.post.controller;

import com.starchive.springapp.post.dto.PostCreateRequest;
import com.starchive.springapp.post.dto.PostListResponse;
import com.starchive.springapp.post.service.PostService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@Tag(name = "게시글")
public class PostController {
    private final PostService postService;

    @PostMapping("/post")
    @Operation(summary = "게시글 작성")
    public ResponseEntity<?> post(@Valid @RequestBody PostCreateRequest request) {

        postService.createPost(request);

        return ResponseEntity.status(201).build();
    }

    @GetMapping("/posts")
    public ResponseEntity<?> findPosts(
            @RequestParam(name = "category", required = false) Long category,
            @RequestParam(name = "tag", required = false) Long tag,
            @RequestParam(name = "page", defaultValue = "0") int page,
            @RequestParam(name = "pageSize", defaultValue = "10") int pageSize) {

        PostListResponse postListResponse = postService.findPosts(category, tag, page, pageSize);

        return ResponseEntity.ok(postListResponse);
    }
}
